### PR TITLE
[REF] web: convert codeEditor tests to HOOT

### PR DESCRIPTION
**Pull request by kebeclibre**
task-3705027

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/core/code_editor.test.js [hunk 1]
```diff
@@ -0,0 +1,276 @@
+/** @odoo-module */
+import { Component, xml, useState, markup } from "@odoo/owl";
+
+import { contains, mountWithCleanup, patchWithCleanup } from "../web_test_helpers";
+import { after, afterEach, expect, getFixture, onError, test } from "@odoo/hoot";
+
+import { CodeEditor } from "@web/core/code_editor/code_editor";
+import { animationFrame } from "@odoo/hoot-mock";
```
Comment (aab-odoo) [inline addons/web/static/tests/core/code_editor.test.js:8]: move above, with hoot imports?

File: addons/web/static/tests/core/code_editor.test.js [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+import { debounce } from "@web/core/utils/timing";
+
+let resizeObserverErrorCount = 0;
+onError((ev) => {
```
Comment (aab-odoo) [inline addons/web/static/tests/core/code_editor.test.js:12]: where does this come from?

File: addons/web/static/tests/core/code_editor.test.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+        static template = xml`<CodeEditor value="state.value" onChange.bind="onChange" />`;
+        static props = ["*"];
+        state = useState({ value: "initial value" });
+        onChange(value) {
+            expect.step(value || "__emptystring__");
+        }
+    }
+
+    const parent = await mountWithCleanup(Parent);
+    expect(".ace_line").toHaveText("initial value");
```
Comment (aab-odoo) [inline addons/web/static/tests/core/code_editor.test.js:155]: we had an extra nextTick before, any idea why?

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/tests/core/code_editor.test.js (hunk 3 above)
Comment (kebeclibre) [reply addons/web/static/tests/core/code_editor.test.js:155]: yes, in mountWithCleanup, by default we wait for or mainComponentContainer to be mounted, which yields
the other animationframe we, in any case, want to wait for

File: addons/web/static/tests/core/code_editor.test.js (hunk 2 above)
Comment (kebeclibre) [reply addons/web/static/tests/core/code_editor.test.js:12]: added a comment

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by aab-odoo** [comment]
Hello @odoo/rd-security 

Could we get an override of the security CI? The spotted markup is used in a test, where we precisely want to have a markuped value to pass as props to a Component.

Thanks

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**